ext/gd/gd.c

```diff
@@ -3560,6 +3560,26 @@ PHP_FUNCTION(imagecrop)
 		RETURN_THROWS();
 	}
 
+	if ((rect.width > 0 && rect.x > INT_MAX - rect.width)) {
```

Comment:
I think we need to check already when assigning to the members of `gdRect` (which are `int`s) above.

And I'm not sure whether allowing negative values makes any sense; from a quick glance, that would be treated like zero. Maybe still support that for 8.3+, but additionally throw for master.

Reply:
I did some [work on master prior](https://github.com/php/php-src/pull/18005), counting on this one before going further.